lib/src/themes/expression/is_supported_script_expression.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:flutter/widgets.dart';
+
+import 'expression.dart';
+
+class IsSupportedScriptExpression extends Expression {
+  final Expression _expression;
+
+  IsSupportedScriptExpression(this._expression)
+      : super('isSupportedScript(${_expression.cacheKey})', _expression.properties());
+
+  @override
+  evaluate(EvaluationContext context) {
+    final operand = _expression.evaluate(context);
+    if (operand is String) {
+      // At the moment, known rendering frontends utilizing this library make use of advanced
```

Comment:
What does this mean? What is "complex support"?

Reply:
I'll update the comment with a Wikipedia link. It's a bit esoteric, but as a native speaker of a language using a Latin-derived script, we naturally assume that, for example, the glyph "a" is always written as "a" but that isn't always true. Complex scripts actually require different rendering of glyphs depending on surrounding context. This goes by a number of names like complex scripts, complex shaping support, complex text layout, etc.: https://en.wikipedia.org/wiki/Complex_text_layout.

The reason that Mapbox / MapLibre have the `is-supported-script` expression is because the "canonical" rendering method for text in that world is such that it can't support complex scripts. Rather each glyph is encoded into a signed distance field which is static by nature and designed to map well into GPU textures for super quick rendering with efficient scale, halo effects, etc. This works super well and has a bunch of cool properties, and even works with Chinese, Japanese, and Korean, but the model totally breaks on a bunch of other (mostly Asian and Indic) scripts where "a" isn't always rendered the same.

(A secondary reason that this `is-supported-script` expression exists is that the JavaScript implementation doesn't support RTL languages like Hebrew and Arabic out-of-the-box, and requires a separate plugin to be loaded... native platforms don't have this issue, so the implementation also needs to account for that.)

I've just pushed an updated comment, but let me know if this requires further explanation or if docs should go in a different place.

Reply:
Thanks for the detailed explanation, that helps a lot.